pkg/roachprod/opentelemetry/cockroachdb_metrics.go

```diff
@@ -0,0 +1,2093 @@
+package opentelemetry
+
+// cockroachdbMetrics is a mapping of CockroachDB metric names to cockroachdb
+// Datadog integration metric names. This allows CockroachDB metrics to comply
+// with the naming requirements for the cockroachdb Datadog integration listed
+// in its metadata.csv.
+// - https://github.com/DataDog/integrations-core/blob/master/cockroachdb/metadata.csv
+// Only metrics whose name matches a key in this map will have its
+// corresponding value exported from the OpenTelemetry Collector.
+var cockroachdbMetrics = map[string]string{
```

Comment:
SRE realizes that maintaining this map is untenable. We are in conversations with Datadog to support the native metric names from CockroachDB which would render this mapping obsolete. Until then we are forced to use this mapping or we can create a custom OpenTelemetry processor to abstract this logic behind.

Reply:
It would have been cleaner to keep the minimum subset, which I counted to be ~81 pairs; the rest is just `s/\./_/g`.

Reply:
I wish I could have! The keys of this map are the metric names as emitted from the CockroachDB Prometheus endpoint (i.e., how OpenTelemetry sees them) and the values are the metric names that the `cockroachdb` Datadog integration expects. That means OpenTelemetry must transform metric names from Prometheus format to Datadog format.

For example, the following entry is one where the `key == value` when the value is `s/\./_/g` but `key != value` when the key is `s/_/\./g`. Since OpenTelemetry will only see the key of this entry the mapping is necessary to do the transformation. I _could_ build more logic into the OpenTelemetry processor but that too would require knowing both the key and the value.

```
"physical_replication_admit_latency": "physical_replication.admit_latency",
```

Additionally, the keys in this map serve as a filter to _only_ allow metrics that appear as keys in the map to be exported to Datadog. That way we don't get billed for custom metrics.